JBPM-6825 - Move saved and quick filters to docks

cristianonicolai opened the pull request.

cristianonicolai commented:
Depends on https://github.com/kiegroup/appformer/pull/225

cristianonicolai commented:
@nmirasch @bsremac please take a look.

bsremac commented on jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/active/ActiveFilterItemRemoved.java:
```diff
@@ -0,0 +1,56 @@
+
+package org.jbpm.workbench.common.client.filters.active;
+
+public class ActiveFilterItemRemoved {
```
ActiveFilterItemRemovedEvent?

cristianonicolai replied:
+1

bsremac commented on jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/active/ActiveFiltersImpl.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Override
+    public HTMLElement getElement() {
```
I perceive this class as a presenter one (although it doesn't carry the name), so it looks a bit strange to me to have it implement _IsElement_.
With that in mind, does _ActiveFilters_ interface need to extend _IsElement_?

cristianonicolai replied:
I agree its a bit different this one, I created as a component that is used in here: https://github.com/cristianonicolai/jbpm-wb/pull/28/files#diff-a5485d38408578707c1e8179af4297ffR75, I guess thats why I was in doubt about calling it presenter. I can rename it but the _IsEleement_ has to stay in order for the Errai injection to work.

bsremac replied:
@cristianonicolai I understand now. I missed the injection point yesterday when commenting.
In that case, I agree that naming the class presenter doesn't really fit its use. 
+1 to leave it as is.

bsremac commented on jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFilterAddEvent.java:
```diff
@@ -0,0 +1,72 @@
+
+    private ColumnFilter filter;
+
+    public BasicFilterAddEvent(final ActiveFilterItem activeFilterItem, final ColumnFilter filter) {
```
The arguments listed in one line.

cristianonicolai replied:
done

cristianonicolai commented:
@nmirasch @bsremac fixed the issue with not listing tasks, it is necessary to include the filter settings columns.

bsremac commented on jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/FilterSettingsManager.java:
```diff
@@ -0,0 +1,45 @@
+
+    void resetDefaultSavedFilters(Consumer<List<SavedFilter>> savedFiltersConsumer);
+
+    FilterSettings getVariablesFilterSettings(String name);
```
How about making this method _default_, if not all implementations will actually override it?

cristianonicolai replied:
+1

bsremac commented on jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListFilterSettingsManager.java:
```diff
@@ -0,0 +1,165 @@
+                + "        \"rowCount\": \"-1\",\n"
+                + "        \"rowOffset\": \"0\",\n";
+        if (name != null) {
+            tableSettingsJSON += "        \"filterOps\":[{\"columnId\":\"" + PROCESS_NAME + "\", \"functionType\":\"EQUALS_TO\", \"terms\":[\"" + name + "\"]}],";
```
Are the preceding spaces intentional?

cristianonicolai replied:
to be honest, I'm not sure, just copied the content over from another place, I would love to replace this code with an actual builder instead of string manipulation. Let me try that.

bsremac commented:
@cristianonicolai  A couple of observations..
When the vertical slider appears on the list table, the elements placed to the right are too close to it. Maybe increasing the space would be a good idea.
![screenshot from 2018-03-08 09-50-40](https://user-images.githubusercontent.com/19285782/37147063-fcabc18c-22c5-11e8-9ba3-ab33ba1618ea.png)

We are still allowing crating filters with the same name. Should we change that in the future?

![screenshot from 2018-03-08 09-51-51](https://user-images.githubusercontent.com/19285782/37147167-4fabe92a-22c6-11e8-80d9-bd61153793e0.png)

cristianonicolai commented:
@bsremac regarding the filter all, that is exactly what the filter "filters on": nothing hehehe. Not sure, perhaps it doesn't sounds like a very useful filter to have by default?

bsremac commented:
@cristianonicolai hahahaha
I think that guy is bound to be retired :)
_Clear All_ gets us there anyway.

cristianonicolai commented:
@bsremac done, I think I got all changes in.
- Removed filter all
- Added validation for filters with same name
- Added 5px padding to list table to avoid getting to close to scroll bar.

bsremac commented:
@cristianonicolai I've just realized that we have _All_ filter in Execution errors, as well :(
